Source/GameBaseFramework/Private/Animation/GBFAnimNotifyState_InputBuffer.cpp

```diff
@@ -0,0 +1,74 @@
+#include "Animation/GBFAnimNotifyState_InputBuffer.h"
+
+#include "GBFLog.h"
+
+#include <Components/SkeletalMeshComponent.h>
+
+void UGBFAnimNotifyState_InputBuffer::NotifyBegin( USkeletalMeshComponent * mesh_component, UAnimSequenceBase * animation, float total_duration, const FAnimNotifyEventReference & event_reference )
+{
+    Super::NotifyBegin( mesh_component, animation, total_duration, event_reference );
+
+    if ( auto * aibc = GetAbilityInputBufferComponent( mesh_component ) )
+    {
+        aibc->StartMonitoring( InputTagsToCheck, TriggerPriority );
+    }
+}
+
+void UGBFAnimNotifyState_InputBuffer::NotifyEnd( USkeletalMeshComponent * mesh_component, UAnimSequenceBase * animation, const FAnimNotifyEventReference & event_reference )
+{
+    Super::NotifyEnd( mesh_component, animation, event_reference );
+
+    if ( auto * aibc = GetAbilityInputBufferComponent( mesh_component ) )
+    {
+        aibc->StopMonitoring();
+    }
+}
+
+UGBFAbilityInputBufferComponent * UGBFAnimNotifyState_InputBuffer::GetAbilityInputBufferComponent_Implementation( const USkeletalMeshComponent * mesh_component ) const
+{
+    if ( mesh_component == nullptr )
+    {
+        return nullptr;
+    }
+
+    UGBFAbilityInputBufferComponent * aibc = nullptr;
```

Comment:
same
I would also do an early return when owner is null and add a log. That way you can initialize the variable here and use auto as well

Reply:
Done